Source/WebCore/platform/text/TextFlags.cpp

```diff
@@ -77,10 +77,10 @@ WTF::TextStream& operator<<(TextStream& ts, FontVariantAlternates alternates)
             append("stylistic(", *values.stylistic, ")");
         if (values.historicalForms)
             append("historical-forms"_s);
-        if (values.styleset)
-            append("styleset(", *values.styleset, ")");
-        if (values.characterVariant)
-            append("character-variant(", *values.characterVariant, ")");
+        if (!values.styleset.isEmpty())
+            append("styleset(", makeStringByJoining(values.styleset, ','), ")");
+        if (!values.characterVariant.isEmpty())
+            append("character-variant(", makeStringByJoining(values.characterVariant, ','), ")");
```

Comment:
A little sad to have to build and discard an intermediate string here. We should seek idioms where we build the string up once, rather than building strings and then building strings from those strings.